lib/ansible/modules/cloud/openstack/os_router.py

```diff
@@ -81,9 +71,20 @@
      default: None
    interfaces:
      description:
-        - List of subnets to attach to the router internal interface.
+        - The router interface  parameters that need to be attached with 
+          the router.Each is a dictionary with the network name or ID(net),
+          subnet name or ID (subnet) and the IP of port (portip) from the
+          network. If no portip  is specified,the default gateway port ip
+          for the subnet will be automatically assigned to port.Assigning 
+          a portip is often  required when a multiple router need to be 
+          connected to a single subnet for which the default gateway has
+          been already used.
```

Comment:
You need to maintain backward compatibility. Your code will need to account for that (which it does not look like it is currently doing).

Reply:
Thanks for your review .  Yes it is indeed not back word compatible since  the interface param has been changed with additional keys , net , subnet and port ip .  I will try to keep the interface param as it is and introduce a new param to capture port ip .   I remember trying to do that but had problem making it compatible with existing logic .   Will  work on this  within couple of week .